uvicorn/server.py

```diff
@@ -112,7 +113,6 @@ def _share_socket(
                 sock_data = sock.share(os.getpid())  # type: ignore[attr-defined]
                 return fromshare(sock_data)
 
-            self.servers = []
```

Comment:
Might want to retain a `self.servers.clear()` so that the list does not have stale items if the server is run multiple times.

Reply:
Should this occur at the end of the `shutdown` method?